### PR TITLE
Alpha wrap

### PR DESCRIPTION
**Pull request by hanywang2**
### Summary 
Add reminaing changes (mostly minor improvements) to wrap CoursePlan's alpha.

Doesn't include a specific major update or additional feature, but a collection of tweaks that enhance the code's efficiency, readability, and more closely resembling the designs.

Changes included but are not limited to styling changes, adding alerts, updating firebase calls, handling edge cases, optimizing code efficiency, adding comments, and cleaning code structure.

This pull request is the first step towards implementing feature CoursePlan's alpha

- [x] changed firebase functions to be called on semesters data change and unified. Works for add, remove, change color, and drag and drop
- [x] added button and function to close onboarding
- [x] styling changes with vue binding to update logout button positioning
- [x] added alerts (providing requirements information and warning on duplicate courses), and hide all but duplicate courses.
- [x] minor styling changes to requirements menu, including temporary descriptions
- [x] update styling for compact view and fix compact view adding courses bug
- [x] prevent duplicates added to the same semester, add a modal with it
- [x] add semesters in chronological order
- [x] add delete semesters menu and confirmations
- [x] update course styling (set height independent of title length or active state), default course color
- [x] update onboarding to make majors dependent on colleges and not require one
- [x] only allow whitelisted users to sign in to alpha
- [x] add Proxima Nova fonts and a favicon
- [x] add more colleges, majors, and minors to requirements
- [x] updated READMEs

### Test Plan 

Screen Shot 2020-01-19 at 7 13 24 PM

Much of the tweaks are accompanied by comments and intended to improve the codebase's readability. If there are unclear functions and declarations, please let me know. I will be significantly less hands-on with the codebase once we are in production.

### CODE REVIEW

File: .firebase/hosting.ZGlzdA.cache [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+img/checkmark.cee94d94.svg,1580354817446,2fec98728f60d6ee9f8a0724c97abd51a1964ede1f5afdd201aed768730df33e
+fonts/ProximaNova-Regular.410504d4.otf,1580354817447,1e1fe09274275f820432d5ec16591663f53463f10f3cad6c383a6252ee3a2704
+fonts/ProximaNova-Extrabold.b4f9eb8c.otf,1580354817446,c83e53616b930684111adbff776e4710d75378b6604d721eecabb87186a4decf
+fonts/ProximaNova-Bold.62d4d7d3.otf,1580354817446,04a122d765539903608456ff08d51dc8deee42baa4a80574e40636b4b2ced3bc
+fonts/ProximaNova-Thin.8f0bc01c.otf,1580354817446,f5fdef059990e98277cf9493fcc62983f8cb16896612619679234126cd296144
+css/chunk-vendors.6f146306.css,1580354817497,f901d7bfa61f0e64bb0152b777eeb7255f257c18a85dcc6dcbdb17ef51a442a7
+js/app.f057e189.js.map,1580354817495,4bdc050b5e3f51624f870b6274e546941d96a85950ef9306db4726afd9dc5acf
+js/app.f057e189.js,1580354817446,308dd0305f199ea53de4bd05aaead06a30491c0c17f8b2c6437cad6be355376b
+js/chunk-vendors.f54b8f75.js,1580354817495,b892b3ebedae60d09a3343858012ce6b856d536c81ad03ed2260a98318304290
+js/chunk-vendors.f54b8f75.js.map,1580354817497,28aa8021c1c4e0e7f84e2fa3362b2a7c39c93b36531546cc1f848a55f2fabe43
```
Comment (SamChou19815) [inline .firebase/hosting.ZGlzdA.cache]: This file should be gitignored

File: src/components/SemesterView.vue [hunk 2]
```diff
@@ -107,6 +121,40 @@ export default {
...
+      }
+    },
+    openSemesterConfirmationModal(type, year, isAdd) {
+      if (isAdd) {
+        this.confirmationText = `Added ${type} ${year} to plan`;
+      } else {
+        this.confirmationText = `Deleted ${type} ${year} from plan`;
+      }
+
+      const confirmationModal = document.getElementById(`semesterConfirmation`);
```
Comment (willespencer) [inline src/components/SemesterView.vue]: This file will need to be heavily edited to remove all of the DOM manipulation.

File: src/components/Requirements.vue [hunk 3]
```diff
@@ -464,20 +493,33 @@ export default {
...
+        // Special search: if search code is not PE or 10XX course
+        if (search === 'all-eligible') return ifAllEligible(courseInfo.subject, courseInfo.catalogNbr.toString());
+        for (const include of includes) {
           for (const option of include) {
+            // Special search: if course code matches code
             if (search === 'code') {
               if (ifCodeMatch(`${courseInfo.subject} ${courseInfo.catalogNbr}`, option)) {
                 // Important: removes array option list from requirements
-                if (includes.length > 1) includes.splice(i, 1);
+                // if (includes.length > 1) includes.splice(i, 1);
```
Comment (willespencer) [inline src/components/Requirements.vue]: Commented line?

File: src/components/Semester.vue [hunk 4]
```diff
@@ -127,158 +168,122 @@ export default {
...
-          console.log('Error getting document:', error);
+    updateBar(course) {
+      this.$emit('updateBar', course);
+    },
+    dragListener(event) {
+      if (!this.$data.scrollable) event.preventDefault();
+    },
+    buildCautions() {
+      this.buildDuplicateCautions();
+      // this.buildDuplicateCautions();
```
Comment (willespencer) [inline src/components/Semester.vue]: Remove comments

File: src/requirements/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# Requirements Data
```
[the remark this answers is not in the document]
Comment (hanywang2) [reply src/requirements/README.md]: no u

File: src/components/SemesterView.vue (hunk 2 above)
Comment (hanywang2) [reply src/components/SemesterView.vue]: okie dokie willie

File: src/components/Dashboard.vue [hunk 6]
```diff
@@ -243,7 +271,6 @@ export default {
         semesters: this.joinOrNAString(course.semesters),
         color: course.color,
         latestSem: course.lastRoster,
-        // requirementsMap: Map,
         id: 1,
```
[the remark this answers is not in the document]
Comment (tcho6319) [reply src/components/Dashboard.vue]: id seems to be unnecessary, so removed!

**Pull request merged**